consul/dnsdisc.py

```diff
@@ -169,26 +172,31 @@ def main():
         token=opts.consul_token,
     )
 
-    if len(opts.query_service) == 0:
-        LOG.error('No service names to query given!')
+    if opts.query_service_name is None:
+        LOG.error('No service name to query given!')
         sys.exit(1)
 
-    services = []
-    for service_name in opts.query_service:
-        LOG.debug('Querying service: %s (%s.%s)',
-                 service_name, opts.query_env, opts.query_stage)
-        services.extend(catalog.all_services(
-            service_name,
-            meta={
-              'env': opts.query_env,
-              'stage': opts.query_stage
-            }
-        ))
+    LOG.debug('Querying service: %s (%s.%s)',
+              opts.query_service_name, opts.query_env, opts.query_stage)
+    services = catalog.all_services(
+        opts.query_service_name,
+        meta={
+            'env': opts.query_env,
+            'stage': opts.query_stage
+        }
+    )
 
     if len(services) == 0:
         LOG.error('No services found!')
         sys.exit(1)
 
+    if opts.query_service_id is not None:
+        LOG.debug('Filtering by ServiceID: %s', opts.query_service_id)
+        services = [
+            x for x in services
+            if x['ServiceID'] == opts.query_service_id
+        ]
```

Comment:
Ah, because we use python lib here, not HTTP api.

It doesn't support filters as I understand:
https://python-consul2.readthedocs.io/en/latest/#consul.base.Consul.Catalog.services

Reply:
I see, that's a good point. Lets leave it as is for now.